### PR TITLE
Geocode js api

### PR DESCRIPTION
**Pull request by tank157**
Big refactor and move over to use the Geocode Javascript API instead of the HTTP one.

Also introduces pools for processing more than 10 elements in a fashion that is Google approved.

### CODE REVIEW

File: src/cljs/gttp2/locations.cljs [hunk 1]
```diff
@@ -46,7 +50,7 @@
           sort-fn (distance-comparator center)
           merged (map-indexed merge-fn civic-data)
           sorted (if center
-                   (sort sort-fn merged)
+                   (sort sort-fn (keep has-geocoded? merged))
```
Comment (tank157) [inline src/cljs/gttp2/locations.cljs]: Whoops, this isn't working like I expect

Comment (tie-rack) [reply src/cljs/gttp2/locations.cljs]: I think you want `filter` here instead of `keep`

Comment (tank157) [reply src/cljs/gttp2/locations.cljs]: Yeah, I'm actually removing the filtering from there for now. I added it to try and fix a problem with the click-drag map functionality, but instead we've just disabled that. So I'm reverting this back to just sorting all the merged locations and will revisit later.

**Comment by tank157** [comment]
@tie-rack If you could give the more recent changes a look over. Not looking for perfection at this point, but any thing I missed would be nice.

**Pull request merged**